test/it/it/suites/DbBaseSpec.scala

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it.suites
+
+
+import cats.effect.{ContextShift, IO, Resource, Timer}
+import com.codahale.metrics.SharedMetricRegistries
+import com.dimafeng.testcontainers.{ForAllTestContainer, PostgreSQLContainer}
+import doobie._
+import doobie.implicits._
+import doobie.postgres.{PFCM, PHC}
+import doobie.util.fragment.Fragment.{const => csql}
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import org.testcontainers.utility.DockerImageName
+import play.api
+import play.api.Application
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
+
+import java.io.File
+import java.nio.file.{Files, StandardOpenOption}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.io.Source
+
+trait DbBaseSpec extends AnyWordSpec with Matchers with BeforeAndAfterAll with ForAllTestContainer with GuiceOneAppPerSuite {
+  implicit val cs: ContextShift[IO] = IO.contextShift(global)
+  implicit val timer: Timer[IO] = IO.timer(global)
+
+  SharedMetricRegistries.clear()
+
+  protected val schemaName = "test_schema"
+
+  override lazy val container: PostgreSQLContainer = PostgreSQLContainer(
+    dockerImageNameOverride = DockerImageName.parse("postgres:10-alpine"),
+    databaseName = "addressbasepremium",
+    username = "root",
+    password = "Passw0rd123"
+  )
+
+  private def createTransactor(): IO[Transactor[IO]] = {
+    val ptx = Transactor.fromDriverManager[IO](
+      driver = "org.postgresql.Driver",
+      url = container.jdbcUrl,
+      user = container.username,
+      pass = container.password
+    )
+
+//    val ptx = Transactor.fromDriverManager[IO](
+//      driver = "org.postgresql.Driver",
+//      url = "jdbc:postgresql://localhost:5433/addressbasepremium",
+//      user = "root",
+//      pass = "password"
+//    )
```

Comment:
Can this comment be removed?  Looks obsolete.

Reply:
Yep - in there so that I can debug locally.

Reply:
Can't we pass in `containt.*` properties on the command line to set the URL, user and pass?

Reply:
We can but then we run the risk of port clashes. This way, TestContainers picks a port randomly and we don't need to care.